Helper/Order.php

```diff
@@ -701,14 +688,60 @@ public function formatReferenceUrl($reference)
     }
 
     /**
-     * Format the (class internal) unambiguous transaction state out of the type and status
+     * Format the (class internal) unambiguous transaction state out of the type and status.
+     * Infer eventually missing states. Represent Bolt partial capture AUTHORIZED state as CAPTURED.
      *
      * @param \stdClass $transaction
+     * @param OrderPaymentInterface $payment
      * @return string
      */
-    public function getTransactionState($transaction)
+    public function getTransactionState($transaction, $payment)
     {
-        return $transaction->type.":".$transaction->status;
+        $transactionState = $transaction->type.":".$transaction->status;
+        $prevTransactionState = $payment->getAdditionalInformation('transaction_state');
+        $prevCaptures = $payment->getAdditionalInformation('captures') ?: [];
+
+        if (!$prevTransactionState) {
+            if (in_array($transactionState, [self::TS_ZERO_AMOUNT, self::TS_CREDIT_COMPLETED])) {
+                return $transactionState;
+            }
+            return self::TS_PENDING;
+        }
+
+        if (in_array($prevTransactionState, [self::TS_PENDING, self::TS_REJECTED_REVERSIBLE]) &&
+            $transactionState == self::TS_AUTHORIZED &&
+            $transaction->captures
+        ) {
+            return self::TS_AUTHORIZED;
+        }
+
+        if (in_array($prevTransactionState, [self::TS_PENDING, self::TS_REJECTED_REVERSIBLE]) &&
+            $transactionState == self::TS_COMPLETED &&
+            count($transaction->captures) > 1
+        ) {
+            return self::TS_AUTHORIZED;
+        }
+
+        if ($prevTransactionState == self::TS_AUTHORIZED &&
+            $transactionState == self::TS_COMPLETED &&
+            count($transaction->captures) - count($prevCaptures) > 1
+        ) {
+            return self::TS_CAPTURED;
+        }
+
+        if ($transactionState == self::TS_AUTHORIZED &&
+            $transaction->captures
+        ) {
+            return self::TS_CAPTURED;
+        }
+
+        if ($prevTransactionState == self::TS_CREDIT_COMPLETED &&
```

Comment:
how is it possible?

Reply:
@daisy1754 
This is where the transaction states transition logic gets complicated. It is not as linear as it was. Consider this flow:
1. PENDING
2. AUTHORIZED
3. partially CAPTURED (<- Magento state), invoice created,  Bolt transaction status is still AUTHORIZED
4. partially or fully refunded previous invoice, CREDIT_COMPLETED state saved to payment in magento, Bolt transaction state is still AUTHORIZED
5. partially CAPTURED again (THIS CASE, which sends us back to #3) or fully CAPTURED (different case, COMPLETED status on both Magento and Bolt), new invoice gets created...